### ISSUE TITLE
line reporting on errors needs adjustment

### ISSUE THREAD

**Issue by lutzhamel** [asteroid-lang/asteroid#41]
The error in this program
```
load "io".
load "util".

structure Person with
  data name.
  data age.
  data gender.
end

-- define a list of persons
let people = [
  Person("George", 32, "M"),
  Person("Steve", 49, "M"),
  Person("Oliver", 21, "M")
].

function f
  with Person(name,age,gender) %if name > 10 do -- error should be reported here
    return 1.
  orwith Person(name,age,gender) %if name == "Steve" do
   return 2.
  end.

let x = f( people @1 ).  -- but error is reported here

println( "The output of function f() is: " + x ).
```
is reported at the call site rather than where the error occurs in the source file.

**Comment by olwmc** [comment]
Working on a fix. Just need to do some more testing to make sure it's robust. Using a similar technique Tim used for #109

**Comment by olwmc** [comment]
Reworking this patch to mirror Tim's solution for #93 . Should be done soon.

**Comment by olwmc** [comment]
Closed with #116
